### PR TITLE
 LABS-767

### PR DESCRIPTION
**Pull request by xlassix**
- [x] create helper create
- [x] Added git action to run test cases 
- [x] Added create_twitter_post_image function
- [x] Create API for the Assistant Plugin
- [x] Added `mint_tweet_request` to lock in mint
- [x] Added test cases for functions added
- [x] Added `cancel_mint_request`
- [x] bitte AI

### CODE REVIEW

File: src/bittle_pluglin/Cargo.toml
Comment (rsoury) [inline src/bittle_pluglin/Cargo.toml:1]: Please please please — no typos, minimise typos. Please rename to `bitte_plugin`

File: src/bittle_pluglin/src/handler/open_api_handler.rs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#[get("/ai-plugin.json")]
+pub async fn open_api_specification() -> Json<Value> {
+  let config = Config::ADDRESS;
+
+  debug!("{}",config);
+
+  println!("Development URL: {}", Config::release_default().address);
+
+  
+    let routes=json!({
```
Comment (rsoury) [inline src/bittle_pluglin/src/handler/open_api_handler.rs:23]: Better to have requested this from a local JSON file directly?

File: src/bittle_pluglin/src/handler/tweet.rs
Comment (rsoury) [inline src/bittle_pluglin/src/handler/tweet.rs:1]: Would be nice to litter some comments throughout and above each method here to describe what it's purpose is, where it's used, etc.

File: src/bittle_pluglin/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+use std::env;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+
+#[launch]
+fn rocket() -> _ {
+    dotenv().expect("Error occured when loading .env");
+
+    env::var("X_CLIENT_ID").expect("X_CLIENT_ID must be set");
+    env::var("TWEET_BEARER").expect("TWEET_BEARER must be set");
```
Comment (rsoury) [inline src/bittle_pluglin/src/main.rs:20]: Are we certain that the Bearer doesn't change?
Should we not request the Bearer via the Oauth endpoint?

File: src/near_nft/nft/src/lib.rs [hunk 3]
```diff
@@ -96,8 +109,102 @@ impl Contract {
...
+        };
+        event.emit();
+
+        (entry.0,entry.1,entry.2)
+    }
+
+    pub fn cancel_mint_request(&mut self, tweet_id: u64) {
+        let tweet_request= self.tweet_requests.get(&tweet_id);
+        if let Some((id,timestamp,_,_))=tweet_request{
+            if (env::block_timestamp_ms()-timestamp)<self.get_lock_time(){
```
Comment (rsoury) [inline src/near_nft/nft/src/lib.rs:148]: I do not believe we want to allow for cancel during the lock time.
If the lock time exceeds, we should allow for cancel AND refund

File: src/bittle_pluglin/src/handler/open_api_handler.rs (hunk 1 above)
Comment (xlassix) [reply src/bittle_pluglin/src/handler/open_api_handler.rs:23]: `servers` in the json is dynamic hence why it couldn't be a static  JSON file

File: src/bittle_pluglin/Cargo.toml
Comment (xlassix) [reply src/bittle_pluglin/Cargo.toml:1]: This was resolved yesterday before our call kindly see #6

File: src/bittle_pluglin/src/main.rs (hunk 2 above)
Comment (xlassix) [reply src/bittle_pluglin/src/main.rs:20]: Yes. The token doesn't change.

File: src/bittle_pluglin/src/handler/tweet.rs
Comment (xlassix) [reply src/bittle_pluglin/src/handler/tweet.rs:1]: On it

File: src/near_nft/nft/src/lib.rs (hunk 3 above)
Comment (xlassix) [reply src/near_nft/nft/src/lib.rs:148]: Previously you started half of the deposit should be refunded if the users cancels before `locktime` set.

File: src/near_nft/nft/src/lib.rs [hunk 4]
```diff
@@ -96,8 +109,102 @@ impl Contract {
...
+    }
+
+    pub fn get_lock_time(&self) -> u64 {
+        self.lock_time
+    }
+
+    #[private]
+    fn claim_funds(&mut self,tweet_id:u64 ) {
+        if let Some((id,_,_,_))= self.tweet_requests.get(&tweet_id){
+            Promise::new(id).transfer(MIN_DEPOSIT/2);
```
Comment (rsoury) [inline src/near_nft/nft/src/lib.rs:191]: This is transferring half here too.

File: src/near_nft/nft/src/lib.rs (hunk 3 above)
Comment (rsoury) [reply src/near_nft/nft/src/lib.rs:148]: Very true.

May be better to consolidate the `claim_funds` and this logic into a single claim_funds that accepts a parameter for the amount of funds to claim.

File: src/bittle_pluglin/Cargo.toml
Comment (rsoury) [reply src/bittle_pluglin/Cargo.toml:1]: It's currently now `bitte_pluglin`. Please make `bitte_plugin`. 

Attention to detail 👀

Comment (xlassix) [reply src/bittle_pluglin/Cargo.toml:1]: Thanks 🥶

**Pull request merged**